### PR TITLE
refactor: replace broadcast mode block with sync and remove fungible params

### PR DESCRIPTION
**Pull request by skosito**
# Description

Part of cosmos sdk update (block mode is deprecated in 0.47) , but also generally better to use sync mode.
Also, removing unused fungible params to make upgrade easier.

Closes: #1930 

## Type of change

Tests

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [x] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [x] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: e2e/txserver/zeta_tx_server.go [hunk 1]
```diff
@@ -188,8 +191,60 @@ func (zts ZetaTxServer) BroadcastTx(account string, msg sdktypes.Msg) (*sdktypes
...
+			return nil, err
+		}
+		return &sdktypes.TxResponse{
+			Code:      res.Code,
+			Codespace: res.Codespace,
+			TxHash:    res.TxHash,
+		}, err
+	}
+
+	var blockTimeout time.Duration = 10 * time.Minute
```
[the remark this answers is not in the document]
Comment (lumtis) [reply e2e/txserver/zeta_tx_server.go:207]: I would say in the `ZetaTxServer` object.
I think we can hardcode a low value in the usage, Cosmos chains are fast and this package is only used in local test for now

File: e2e/txserver/zeta_tx_server.go [hunk 2]
```diff
@@ -188,8 +191,60 @@ func (zts ZetaTxServer) BroadcastTx(account string, msg sdktypes.Msg) (*sdktypes
...
+		case <-time.After(time.Millisecond * 100):
+			resTx, err := zts.clientCtx.Client.Tx(context.TODO(), hash, false)
+			if err == nil {
+				txRes, err := mkTxResult(zts.clientCtx, resTx)
+				if err == nil {
+					return txRes, nil
+				}
+			}
+		}
+	}
```
Comment (lumtis) [inline e2e/txserver/zeta_tx_server.go:226]: I think we can create a separate function

File: changelog.md [hunk 3]
```diff
@@ -59,6 +59,7 @@
 * [1941](https://github.com/zeta-chain/node/pull/1941) - add unit tests for zetabridge package
 * [1985](https://github.com/zeta-chain/node/pull/1985) - improve fungible module coverage
 * [1992](https://github.com/zeta-chain/node/pull/1992) - remove setupKeeper from crosschain module
+* [2001](https://github.com/zeta-chain/node/pull/2001) - replace broadcast mode block with sync
```
Comment (lumtis) [inline changelog.md:62]: We change cmd root, I think we can consider it a refactoring

**Review by lumtis — COMMENTED** [review]
There is a lint issue apparently

**Review by lumtis — APPROVED** [review]
Looks good, can we rename the PR title?

File: proto/fungible/genesis.proto [hunk 4]
```diff
@@ -2,15 +2,13 @@ syntax = "proto3";
...
 import "gogoproto/gogo.proto";
 
 option go_package = "github.com/zeta-chain/zetacore/x/fungible/types";
 
 // GenesisState defines the fungible module's genesis state.
 message GenesisState {
-  Params params = 1 [(gogoproto.nullable) = false];
-  repeated ForeignCoins foreignCoinsList = 2 [(gogoproto.nullable) = false];
-  SystemContract systemContract = 3;
+  repeated ForeignCoins foreignCoinsList = 1 [(gogoproto.nullable) = false];
```
Comment (ws4charlie) [inline proto/fungible/genesis.proto:12]: I think we should keep existing field indexes `2` and `3` untouched. Changing field indexes will probably break backward compatibility in `protobuf` context (for instance, the queries to genesis state). @kingpinXD  correct me if I'm wrong.

File: x/fungible/genesis.go [hunk 5]
```diff
@@ -18,14 +18,12 @@ func InitGenesis(ctx sdk.Context, k keeper.Keeper, genState types.GenesisState)
 		k.SetSystemContract(ctx, *genState.SystemContract)
 	}
 
-	k.SetParams(ctx, genState.Params)
```
Comment (ws4charlie) [inline x/fungible/genesis.go:21]: Do we plan to clean up this `Params` from chain state thru migration script or forget about it?

**Review by kingpinXD — APPROVED** [review]

**Review by lumtis — APPROVED** [review]

**Pull request merged**